src/compiler/sourcemap.ts

```diff
@@ -0,0 +1,416 @@
+/// <reference path="checker.ts"/>
+
+/* @internal */
+namespace ts {
+    export interface SourceMapWriter {
+        getSourceMapData(): SourceMapData;
+        setSourceFile(sourceFile: SourceFile): void;
+        emitPos(pos: number): void;
+        emitStart(range: TextRange): void;
+        emitEnd(range: TextRange): void;
+        pushScope(scopeDeclaration: Node, scopeName?: string): void;
+        popScope(): void;
+        getText(): string;
+        getSourceMappingURL(): string;
+        initialize(filePath: string, sourceMapFilePath: string, sourceFiles: SourceFile[], isBundledEmit: boolean): void;
+        reset(): void;
+    }
+
+    const nop = <(...args: any[]) => any>Function.prototype;
+    let nullSourceMapWriter: SourceMapWriter;
+
+    export function getNullSourceMapWriter(): SourceMapWriter {
+        if (nullSourceMapWriter === undefined) {
+            nullSourceMapWriter = {
+                getSourceMapData: nop,

```

Comment:
this needs to be checked, but AFAIR calling vararg function with different number of arguments is not a very good pattern for V8. When source map code was written initially it has the same shape but then it was rewritten so each nop function has the same number of arguments with the corresponding method in the interface.